mlir/unittests/Analysis/Presburger/MatrixTest.cpp

```diff
@@ -248,3 +248,85 @@ TEST(MatrixTest, computeHermiteNormalForm) {
     checkHermiteNormalForm(mat, hermiteForm);
   }
 }
+
+TEST(MatrixTest, inverse) {
+  FracMatrix mat = makeFracMatrix(
+      2, 2,
+      {{Fraction(2), Fraction(1)},
+       {Fraction(7), Fraction(0)}});
+  FracMatrix inverse = makeFracMatrix(
+      2, 2,
+      {{Fraction(0), Fraction(1, 7)},
+       {Fraction(1), Fraction(-2, 7)}});
+
+  FracMatrix inv(2, 2);
+  mat.determinant(&inv);
+
+  for (unsigned row = 0; row < 2; row++)
+    for (unsigned col = 0; col < 2; col++)
+      EXPECT_EQ(inv(row, col), inverse(row, col));
+  
+  mat = makeFracMatrix(
+      2, 2,
+      {{Fraction(0), Fraction(1)},
+       {Fraction(0), Fraction(2)}});
+  Fraction det = mat.determinant(nullptr);
+  
+  EXPECT_EQ(det, Fraction(0));
+
+  mat = makeFracMatrix(
+      3, 3,
+      {{Fraction(1), Fraction(2), Fraction(3)},
+       {Fraction(4), Fraction(8), Fraction(6)},
+       {Fraction(7), Fraction(8), Fraction(6)}});
+  inverse = makeFracMatrix(
+      3, 3,
+      {{Fraction(0),     Fraction(-1, 3), Fraction(1, 3)},
+       {Fraction(-1, 2), Fraction(5, 12), Fraction(-1, 6)},
+       {Fraction(2, 3),  Fraction(-1, 6), Fraction(0)}});
+
+  mat.determinant(&inv);
+  for (unsigned row = 0; row < 3; row++)
+    for (unsigned col = 0; col < 3; col++)
+      EXPECT_EQ(inv(row, col), inverse(row, col));
+
+  mat = makeFracMatrix(0, 0, {});
+  mat.determinant(&inv);
+}
+
+TEST(MatrixTest, intInverse) {
+  IntMatrix mat = makeIntMatrix(2, 2, {{2, 1}, {7, 0}});
+  IntMatrix inverse = makeIntMatrix(2, 2, {{0, -1}, {-7, 2}});
+
+  IntMatrix inv(2, 2);
+  mat.determinant(&inv);
+
+  for (unsigned i = 0; i < 2u; i++)
+    for (unsigned j = 0; j < 2u; j++)
+      EXPECT_EQ(inv(i, j), inverse(i, j));
+
```

Comment:
I would suggest just adding a Matrix operator== and then using EXPECT_EQ on the matrices here. Either that or at least make a function to EXPECT_EQ two matrices (and their dimensions)